src/main/java/jenkins/scm/api/SCMNavigatorDescriptor.java

```diff
@@ -40,6 +40,12 @@ protected SCMNavigatorDescriptor(Class<? extends SCMNavigator> clazz) {
         super(clazz);
     }
 
+    public abstract String getDescription();
+
+    public abstract String getCategoryId();
```

Comment:
:bug: (3×) should probably be marked `@Nullable` in that a non-null return value is requested only if also returning non-null from `newInstance` (whose Javadoc should also be updated to note that creating a `newInstanceDetail` view is unnecessary if you implement `getDescription`).